scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -766,16 +766,10 @@ static s32 pick_idle_cpu_from(const struct cpumask *cand_cpumask, s32 prev_cpu,
 	 * partially idle @prev_cpu.
 	 */
 	if (smt_enabled) {
-
-		// if layer is prev_over_idle_core, try prev
-		if (prev_in_cand && layer->prev_over_idle_core &&
-				scx_bpf_test_and_clear_cpu_idle(prev_cpu))
-				return prev_cpu;
-
-		// try prev if layers smt sibling is idle
-		if (prev_in_cand &&
-		    bpf_cpumask_test_cpu(prev_cpu, idle_smtmask) &&
-		    scx_bpf_test_and_clear_cpu_idle(prev_cpu))
+		// use prev if sibling is free or prev_over_idle_core
+		if (prev_in_cand && (layer->prev_over_idle_core
+			|| bpf_cpumask_test_cpu(prev_cpu, idle_smtmask)) &&
+			scx_bpf_test_and_clear_cpu_idle(prev_cpu))
```

Comment:
`||` being on the previous line would be more consistent. I'd just put each condition on its own line with the right indentation. It's a bit difficult to read as-is.

Reply:
Also, below, there's another sometimes duplicate `scx_bpf_test_and_clear_cpu_idle(prev_cpu)` right after the `if (smt_enabled) {}` block. We can skip that if `smt_enabled && prev_over_idle_core`, right? An easier way to structure this may be something like this:
```
if (prev_in_cand && prev_over_idle_core) {
    if (test_idle())
        return prev_cpu;
    prev_in_cand = false;
}

if (smt_enabled) {
    if (prev_in_cand && test_smt_mask() && test_idle())
            return prev_cpu;
   ...
}

if (prev_in_cand && test_idle())
    return prev_cpu; 
```

Reply:
>     prev_in_cand = false;

ahh that's kinda cool.

Reply:
I *thinK* that gets rid of the possible duplicate calls while still doing the right thing?